plugins/azuread/pkg/fetch/fetch.go

```diff
@@ -70,38 +50,74 @@ func (f *Fetcher) Fetch(ctx context.Context, outputWriter, errorWriter io.Writer
 		}
 	}
 
+	for user, err := range f.fetchUsers(ctx) {
+		if err != nil {
+			common.WriteErrorWithExitCode(errorWriter, err, 1)
+		}
+
+		if err := jsonWriter.Write(user); err != nil {
+			_, _ = errorWriter.Write([]byte(err.Error()))
+		}
+	}
+
+	return nil
+}
+
+func (f *Fetcher) fetchUsers(ctx context.Context) iter.Seq2[map[string]any, error] {
 	aadUsers, err := f.azureClient.ListUsers(ctx, f.Groups, f.userProps)
 	if err != nil {
-		common.WriteErrorWithExitCode(errorWriter, err, 1)
+		return fetcher.YieldError(err)
 	}
 
-	for _, user := range aadUsers {
+	return fetcher.YieldMap(func(object any) ([]byte, error) {
+		user, ok := object.(models.Userable)
```

Comment:
If `models.Userable` is an interface, you can replace this runtime type assertion with a compile-time type constraint by defining `YieldMap` as:
```go
func YieldMap[T any](objects []T, marshaller func(T) ([]byte, error)) iter.Seq2[map[string]any, error] {
    ...
}
```

I also swapped the order of arguments to make the code easier to read and write. It's really hard to see the second argument when the first one is a long anonymous function.

Reply:
What's the type of `aadUsers`? Why can't the callback function take an argument of that type instead of `any`?